bulbs/poll/models.py

```diff
@@ -0,0 +1,235 @@
+            'access_token': vault.read('sodahead/token')['value'],
+            'name': self.title,
+            'title': self.question_text,
+        }
+
+        if self.sodahead_id:
+            payload['id'] = self.sodahead_id
+
+        if self.published:
+            activation_date = self.published.astimezone(pytz.utc)
+            payload['activationDate'] = activation_date.strftime(SODAHEAD_DATE_FORMAT)
+        else:
+            payload['activationDate'] = None
+
+        if self.end_date:
+            end_date = self.end_date.astimezone(pytz.utc)
+            payload['endDate'] = end_date.strftime(SODAHEAD_DATE_FORMAT)
+        else:
+            payload['endDate'] = ''
+
+        for answer in self.answers.all():
+            if answer.answer_text:
+                payload[answer.sodahead_answer_id] = BLANK_ANSWER
+            else:
+                payload[answer.sodahead_answer_id] = answer.answer_text
+
+        if 'answer_01' not in payload:
+            payload['answer_01'] = DEFAULT_ANSWER_1
+
+        if 'answer_02' not in payload:
+            payload['answer_02'] = DEFAULT_ANSWER_2
+
+        return payload
+
+    def save(self, *args, **kwargs):
+        if not self.sodahead_id:
+            response = requests.post(SODAHEAD_POLLS_ENDPOINT, self.sodahead_payload())
+            if response.status_code > 499:
+                raise SodaheadResponseError(response.text)
+            elif response.status_code > 399:
+                raise SodaheadResponseFailure(response.text)
+            else:
+                self.sodahead_id = response.json()['poll']['id']
+        else:
+            self.sync_sodahead()
+
+        super(Poll, self).save(*args, **kwargs)
+
+    def delete(self, *args, **kwargs):
+        response = requests.delete(
+            SODAHEAD_DELETE_POLL_ENDPOINT.format(
+                self.sodahead_id,
+                vault.read('sodahead/token')['value'],
+            )
+         )
+        if response.status_code > 499:
+            raise SodaheadResponseError(response.text)
+        elif response.status_code > 399:
+            raise SodaheadResponseFailure(response.text)
+        super(Poll, self).delete(*args, **kwargs)
+
+    def sync_sodahead(self):
+        response = requests.post(
+            SODAHEAD_POLL_ENDPOINT.format(self.sodahead_id),
+            self.sodahead_payload()
+        )
+
+        if response.status_code > 499:
+            raise SodaheadResponseError(response.json())
+        elif response.status_code > 399:
+            raise SodaheadResponseFailure(response.json())
+
+
+class Answer(Indexable):
+    poll = models.ForeignKey(
+        Poll,
+        on_delete=models.CASCADE,
+        related_name='answers'
+    )
+    sodahead_answer_id = models.CharField(max_length=20, blank=True, default="")
+    answer_text = models.TextField(blank=True, default="")
+
+    def save(self, *args, **kwargs):
+        # using transaction/select_for_update here because we don't want to
```

Comment:
I don't have a preference for docstrings vs multiline comments, and I just silence lint errors about missing docstrings. 
Any reason we should have a preference?